src/compiler/binder.ts

```diff
@@ -1787,6 +1796,15 @@ namespace ts {
             switch (node.kind) {
                 /* Strict mode checks */
                 case SyntaxKind.Identifier:
+                    // for typedef type names with namespaces

```

Comment:
As you explained to me in person, the declaration can only be bound here because all the containing namespace declarations will be in effect, which lets the typedef fall into the right namespace. Can you change the comment to explain this?